Dockerfile

```diff
@@ -9,13 +9,18 @@ WORKDIR /app
 COPY poetry.lock pyproject.toml ./
 RUN poetry install --without dev --no-root && rm -rf "$POETRY_CACHE_DIR"
 
+FROM python:3.12-bookworm as version
+COPY .git .
+RUN git describe --tags --abbrev=0 > VERSION
```

Comment:
What happens when the current HEAD doesn't have a tag? 25.0.1-abc123?, where abc123 is short commit hash?

Reply:
So it depends, if only the current version has been fetch and it lacks a tag then the checkout fails since it needs to find a tag somewhere in the history. Otherwise it will just grab the most recent tag. 

When on a version with a tag, it is just the current tag. `eg. 25.0.1` 

When the current HEAD doesn't have a tag, but one exist earlier in the commit history, it will give the most recent tag currently w/o the short hash. `eg. 25.0.1`  Removing  `--abbrev=0` makes it something like `24.1.0-6-g58b98e072` which matches the style for a pre-release [semver rule 9.](https://semver.org/#spec-item-9) but this may end up appearing in release builds.

The best option for including short hashes might be to go back to using the release tag from GH for release builds, then use this in a non-release context

Reply:
Updated:  The tag for non-head commits is now formatted as: `24.1.0-pre+g58b98e072`